app/entities/user.go

```diff
@@ -13,5 +18,127 @@ type User struct{}
 
 // Validate will validate User's signup data
 func (u User) Validate() error {
+	if err := checkName(u.Firstname); err != nil {
+		return err
+	}
+	if u.Lastname != "" {
+		if err := checkName(u.Lastname); err != nil {
+			return err
+		}
+	}
+
+	if err := checkMail(u.Email); err != nil {
+		return err
+	}
+
+	if err := checkPass(u.Password); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkName(name string) error {
+	const (
+		minNameLen         int    = 2   // 2
+		maxNameLen         int    = 256 // 256
+		invalidNameSymbols string = "!\"#$%&*+,./:;<=>?@[\\]^_`{|}~1234567890"
+	)
+	// Checking the length
+	nameRune := []rune(name)
+	nameLen := len(nameRune)
+	switch {
+	case nameLen < minNameLen:
+		return errors.New("Name is too short.\n")
+	case nameLen > maxNameLen:
+		return errors.New("Name is too long.\n")
+	}
+	// Checking for not allowed symbols
+	invalidNameSymbolsRune := []rune(invalidNameSymbols)
+	for _, v := range nameRune {
```

Comment:
It's too complex algorithm.
We should always try to avoid loop in loop if that is possible.

As alternative, it looks like we can use `regular expressions` here.

Reply:
Регулярні вирази сиджу вивчаю. Руни вже виключені в останньому коміті.